### DISCUSSION TITLE
[tree chart] Add option to expand N levels and expand all

### DISCUSSION THREAD

**Discussion by kamalkeshavani-aiinside** [apache/superset#18520] [Ideas]
**Is your feature request related to a problem? Please describe.**
Current configuration of Tree chart expands to only 3 levels by default. Then user needs to click on bottom levels(nodes), to expand the tree further.

**Describe the solution you'd like**
In customize tab of explore view, an option should allow to select default expand levels to a number N or all.

**Describe alternatives you've considered**

**Additional context**

**Comment by junlincc** [comment]
looping in the viz owner @mayurnewase :)

**Comment by mayurnewase** [comment]
Hey Kamal thanks for raising this.
It just need this parameter from control panel.
![image](https://user-images.githubusercontent.com/12967587/139452672-650cb8dc-20dd-4237-8f11-45b37b13ae2d.png)

Would you like to open PR,or should I do it?

**Comment by kamalkeshavani-aiinside** [comment]
@mayurnewase Thank you for the response.
Please raise the PR.

**Comment by cbhoga** [comment]
Hello, where to configure series-tree.initialTreeDepth. Please advise as tree in superset showing only 2 levels but customer need complete tree to be visible.

**Reply by Asifalibn** [reply]
~/superset-master/superset-frontend/plugins/plugin-chart-echarts/src/Tree/transformProps.ts in const series: add initialTreeDepth

**Comment by SBIN2010** [comment]
Implemented in #35425
